src/containers/barray.mli

```diff
@@ -0,0 +1,212 @@
+
+(* This file is free software, part of containers. See file "license" for more details. *)
+
+type 'a t = 'a array
+
+val empty : 'a t
+
+val isEmpty: 'a t -> bool
+
+external make : int -> 'a -> 'a t = "caml_make_vect"
+(** [Barray.make n x] returns a fresh t of length [n],
+    initialized with [x].
+    All the elements of this new t are initially
+    physically equal to [x] (in the sense of the [==] predicate).
+    Consequently, if [x] is mutable, it is shared among all elements
+    of the t, and modifying [x] through one of the t entries
+    will modify all other entries at the same time.
+    Raise [Invalid_argument] if [n < 0] or [n > Sys.max_array_length].
+    If the value of [x] is a floating-point number, then the maximum
+    size is only [Sys.max_array_length / 2].*)
+
+val makeWithInit : int -> (int -> 'a) -> 'a t
+(** [Barray.init n f] returns a fresh t of length [n],
+    with element number [i] initialized to the result of [f i].
+    In other terms, [Array.init n f] tabulates the results of [f]
+    applied to the integers [0] to [n-1].
+    Raise [Invalid_argument] if [n < 0] or [n > Sys.max_array_length].
+    If the return type of [f] is [float], then the maximum
+    size is only [Sys.max_array_length / 2].*)
+
+val length : 'a t -> int
+
+val get : int -> 'a t -> 'a option
+
+val set : int -> 'a -> 'a t -> bool
+
+val getOrRaise : int -> 'a t -> 'a
+
+val setOrRaise : int -> 'a -> 'a t -> unit
+
+val unsafeGet : int -> 'a t -> 'a
+
+val unsafeSet : int -> 'a -> 'a t -> unit
```

Comment:
I don't have a strong opinion on whether or not to include these, but if they're included I'd like them to be named `unsafeGetUnchecked` and `unsafeSetUnchecked`